### ISSUE TITLE
"internal error: expectJust getPackageDetails" while processing Cabal-1.18.1.6

### ISSUE THREAD

**Issue by peti** [haskell/haddock#442]
This happens with the Haddock version included in ghc-7.11.20150809. See http://hydra.cryp.to/build/1115397/log/raw for the full detail.

**Comment by Fuuzetsu** [comment]
See #363 ; I don't see why you'd only start hitting it now however unless nixpkgs suddenly started using older Cabal than it has been.

**Comment by peti** [comment]
So, do I get this right that the error is going to go away once we update to a newer GHC snapshot?

**Comment by Fuuzetsu** [comment]
Well,l I don't know. The fix has been in Cabal since like March. To best of my knowledge, it has been included with the Cabal that came with 7.10.1, even the RC3, at least as far as I'm reading https://github.com/haskell/cabal/pull/2439 right. Based on this alone I don't think updating to more recent snapshot will be of any help.

In the log I see

```
Using Cabal-1.9999 compiled by ghc-7.11
Using compiler: ghc-7.11.20150809
```

The question is what is the actual Cabal version being used. I have added `--package-name` and `--package-version` flags in Haddock which allow cabal-install to invoke Haddock with the information it needs. If those flags are not present then it tries asking Cabal. If that info is not available to Cabal, we get a crash (and this crash is inside Cabal, not Haddock). See comment on `modulePackageInfo` in `haddock-api/src/Haddock.hs` for few more words on the matter.

In the end there is nothing I can do to help you.The problem is either cabal-install/Cabal being too old or to new and not invoking Haddock right but you'd have to ask the cabal folk for details. Maybe @23Skidoo knows something and can point you in the right direction. Also @bgamari .

Really you should start by finding out what Cabal version you are using first, preferably which commit.

**Comment by peti** [comment]
OK. I'll take a look at this, and if I find out something that suggests this may be bug in Haddock,then I'll re-open this ticket. Thank you very much for your efforts!
